src/App.tsx

```diff
@@ -13,20 +13,51 @@ interface AppProps {
 function App(props: AppProps) {
   let basicCards: JSX.Element[] = [];
 
-  props.basicCards && props.basicCards.forEach((cardData, cardIndex) => {
-    basicCards.push(<Card faceValue={cardData.face} flipValue={cardData.back} key={cardIndex+1} />)
-  });
+  const [cardsArray, setCardsArray] = useState(basicCards);
+  const [incrementalKey, setIncrementalKey ] = useState(props.basicCards?.length || 0)
+
+  useEffect(() => {
+    setIncrementalKey(prevKey => prevKey +1)
+  }, [cardsArray, setCardsArray])
```

Comment:
This effect sometimes causes the key amount to be bigger than the total amount of cards created, but I don't think that's an issue because the most important part is that they are always different, and that the newly created cards always have key value bigger than previous cards.